src/SimulationRunner.cc

```diff
@@ -572,13 +572,15 @@ void SimulationRunner::UpdateSystems()
 
   {
     IGN_PROFILE("PostUpdate");
+    this->entityCompMgr.LockAddingEntitiesToViews(true);
     // If no systems implementing PostUpdate have been added, then
     // the barriers will be uninitialized, so guard against that condition.
     if (this->postUpdateStartBarrier && this->postUpdateStopBarrier)
     {
       this->postUpdateStartBarrier->Wait();
       this->postUpdateStopBarrier->Wait();
     }
+    this->entityCompMgr.LockAddingEntitiesToViews(false);
   }
```

Comment:
I know that server system `PostUpdate`s are ran in parallel (i.e., across multiple threads), but what about GUI systems? Do any GUI system `*Update` methods run in parallel as well?

Reply:
I think the gui system updates do not run in parallel:
https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo5/src/gui/GuiRunner.cc#L199-L202

Reply:
Great, thanks for the link! I will go ahead and mark this as resolved, since it looks like we only need to do this for server systems.